File: C++ Programs/Palindrome-string.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    int isPalindrome(string S) {
        int n = S.size();

        for (int i = 0; i < n / 2; i++)
            if (S[i] != S[n - i - 1])
                return 0;
        return 1;
    }
};

int main() {
    Solution solution;
    string input;

    cout << "Enter a string: ";
    cin >> input;

    if (solution.isPalindrome(input)) {
        cout << "The string is a palindrome." << endl;
    } else {
        cout << "The string is not a palindrome." << endl;
    }

    return 0;
}

File: Python Programs/Palindrome-string.py
def is_palindrome(s):
    # Remove spaces and convert to lowercase to make it case-insensitive
    s = s.replace(" ", "").lower()
    
    # Compare the original string with its reverse
    return s == s[::-1]

# Sampple test case
input_string = input("Enter a string: ")
if is_palindrome(input_string):
    print(f"'{input_string}' is a palindrome.")
else:
    print(f"'{input_string}' is not a palindrome.")
